src/Pact/Eval.hs

```diff
@@ -233,6 +233,28 @@ evalNamespace info setter m = do
         Nothing -> ModuleName nn (Just n)
         Just {} -> mn
 
+-- | Lookup module in state or database with exact match on 'ModuleName'.
+lookupModule :: HasInfo i => i -> ModuleName -> Eval e (Maybe (ModuleData Ref))
+lookupModule i mn = do
+  loaded <- preuse $ evalRefs . rsLoadedModules . ix mn
+  case loaded of
+    Just (m,_) -> return $ Just m
+    Nothing -> do
+      stored <- readRow (getInfo i) Modules mn
+      case stored of
+        Just mdStored -> do
+          natives <- view $ eeRefStore . rsNatives
+          let natLookup (NativeDefName n) = case HM.lookup (Name n def) natives of
+                Just (Direct t) -> Just t
+                _ -> Nothing
+          case traverse (traverse (fromPersistDirect natLookup)) mdStored of
+            Right md -> do
+              evalRefs . rsLoadedModules %= HM.insert mn (md,False)
```

Comment:
reinserting with `(md,False)`? Should this preserve what the old value of `snd` would be?

Reply:
No, the only cases in which the flag is `True` is if the module was newly-loaded or updated. Here we're reinflating it from the db so it is by construction `False` as we're not modifying it.